package/libskylark/src/libskylark.c

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (C) 2016 Swift Navigation Inc.
+ * Contact: Jacob McNamee <jacob@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <curl/curl.h>
+#include <libsbp/observation.h>
+#include <libsbp/navigation.h>
+#include <string.h>
+#include <stdio.h>
+//#include <sbp_settings.h>
+#include <sbp_zmq.h>
+
+#include "libskylark.h"
+
+#define VERBOSE
+
+/**
+ *  Utilities
+ */
+
+// TODO (mookerji): Repeated here as is everywhere else! This needs to be moved
+// into a piksi buildroot library.
+static int file_read_string(const char *filename, char *str, size_t str_size)
+{
+  FILE *fp = fopen(filename, "r");
+  if (fp == NULL) {
+    log_error("error opening %s\n", filename);
+    return -1;
+  }
+  bool success = (fgets(str, str_size, fp) != NULL);
+  fclose(fp);
+  if (!success) {
+    log_error("error reading %s\n", filename);
+    return -1;
+  }
+  return 0;
+}
+
+// TODO (mookerji): Repeated here as is everwhere else! This needs to be moved
+// into a piksi buildroot library.
+static RC get_sbp_sender_id(u16* sender_id)
+{
+  *sender_id = SBP_SENDER_ID;
+  char sbp_sender_id_string[32];
+  if (file_read_string(FILE_SBP_SENDER_ID, sbp_sender_id_string,
+                       sizeof(sbp_sender_id_string)) < 0) {
+    return E_GENERIC_ERROR;
+
+  }
+  *sender_id = strtoul(sbp_sender_id_string, NULL, 10);
+  return NO_ERROR;
+}
+
+/**
+ *  Configuration
+ */
+
+void log_client_config(const client_config_t *config)
+{
+  log_debug("client_config: endpoint_url=%s\n", config->endpoint_url);
+  log_debug("client_config: accept_type_header=%s\n", config->accept_type_header);
+  log_debug("client_config: content_type_header=%s\n", config->content_type_header);
+  log_debug("client_config: encoding=%s\n", config->encoding);
+  log_debug("client_config: device_header=%s\n", config->device_header);
+  log_debug("client_config: sbp_sender_id=%d\n", config->sbp_sender_id);
+  log_debug("client_config: enabled=%d\n", config->enabled);
+  log_debug("client_config: fd=%d\n", config->fd);
+}
+
+int client_config_compare(const client_config_t *a,
+                          const client_config_t *b)
+{
+  return strcmp(a->endpoint_url, b->endpoint_url) == 0
```

Comment:
Is this going to be used? it's mixing some user config with running config so might not be the right function to use.